### PR TITLE
Fix wrong checkpoint info after restart

### LINKED ISSUE #7799: RW Disaggregate with s3 tiflash crash for Unsupport type: Invalid
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

rw disaggregated with ks3, consistency case
tiflash crash
```
[2023/07/12 14:14:33.136 +08:00] [ERROR] [BaseDaemon.cpp:368] ["(from thread 99) Terminate called after throwing an instance of DB::Exception\nCode: 49, e.displayText() = DB::Exception: Unsupport type: Invalid, e.what() = DB::Exception\nStack trace:\n\n\n       0x759827d\tterminate_handler() [tiflash+123306621]\n                \tlibs/libdaemon/src/BaseDaemon.cpp:633\n  0x7fa27fe9fa13\tstd::__terminate(void (*)()) [libc++abi.so.1+236051]\n  0x7fa27fe9f9b8\tstd::terminate() [libc++abi.so.1+235960]\n       0x1c3672b\t__clang_call_terminate [tiflash+29583147]\n       0x7e6c853\tstd::__1::__packaged_task_func<DB::DM::Remote::DataStoreS3::getDataFilesInfo(std::__1::unordered_set<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::hash<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::equal_to<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&)::$_3,"] [source=BaseDaemon] [thread_id=327]
[2023/07/12 14:14:33.136 +08:00] [ERROR] [BaseDaemon.cpp:376] [########################################] [source=BaseDaemon] [thread_id=327]
[2023/07/12 14:14:33.136 +08:00] [ERROR] [BaseDaemon.cpp:377] ["(from thread 99) Received signal Aborted(6)."] [source=BaseDaemon] [thread_id=327]
```
### 4. What is your TiFlash version? (Required)
TiFlash
Release Version: v7.3.0-alpha
Edition:         Community
Git Commit Hash: c6944e7e01ac592fdeb189b490196c967f03a02b
Git Branch:      heads/refs/tags/v7.3.0-alpha
UTC Build Time:  2023-07-11 11:17:12
Enable Features: jemalloc sm4(GmSSL) avx2 avx512 unwind thinlto
Profile:         RELWITHDEBINFO

Raft Proxy
Git Commit Hash:   bc0ebe45a0183c0afb67db609157119776210d5e
Git Commit Branch: HEAD
UTC Build Time:    2023-07-11 11:21:54
Rust Version:      rustc 1.67.0-nightly (96ddd32c4 2022-11-14)
Storage Engine:    tiflash
Prometheus Prefix: tiflash_proxy_
Profile:           release
Enable Features:   Unknown (env var does not exist when building)

### PR DESCRIPTION
**Pull request by lidezhu**
### What problem does this PR solve?

Issue Number: close #7799 

Problem Summary: After disable uploading page data when do checkpoint, the data file id in manifest will be empty. And after restart, tiflash may try to recover from the latest manifest file and set a empty checkpoint location which is unexpected and will make tiflash crash in the next round of uploading checkpoint.

### What is changed and how it works?
When restoring from manifest file at restart, check whether the data file id is empty before copy the checkpoint info to memory.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
1. deploy a cluster and load some data;
2. restart and write more data, and then check whether tiflash will crash;
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by Lloyd-Pottiger — APPROVED** [review]

File: dbms/src/Storages/Page/V3/PageDirectory.cpp [hunk 1]
```diff
@@ -531,7 +531,8 @@ void VersionedPageEntries<Trait>::copyCheckpointInfoFromEdit(const typename Page
 
     // Pre-check: All ENTRY edit record must contain checkpoint info for copying.
     RUNTIME_CHECK(edit.type == EditRecordType::VAR_ENTRY);
-    RUNTIME_CHECK(edit.entry.checkpoint_info.has_value());
+    if (!edit.entry.checkpoint_info.has_value())
```
Comment (JaySon-Huang) [inline dbms/src/Storages/Page/V3/PageDirectory.cpp:534]:
```suggestion
    // The checkpoint_info from `edit` could be empty when we upload the manifest without any page data
    if (!edit.entry.checkpoint_info.has_value())
```

File: dbms/src/Storages/Page/V3/PageDirectory.cpp [hunk 2]
```diff
@@ -531,7 +531,8 @@ void VersionedPageEntries<Trait>::copyCheckpointInfoFromEdit(const typename Page
 
     // Pre-check: All ENTRY edit record must contain checkpoint info for copying.
```
Comment (JaySon-Huang) [inline dbms/src/Storages/Page/V3/PageDirectory.cpp:532]:
```suggestion
```

**Review by JaySon-Huang — APPROVED** [review]
LGTM with small comment

**Pull request merged**